var/spack/repos/builtin/packages/sensei/package.py

```diff
@@ -41,20 +43,23 @@ class Sensei(CMakePackage):
     # Paraview 6 cannot be used since it requires Python 3. Starting from
     # version 3, SENSEI supports Python 3.
     depends_on("paraview@5.5.0:5.5.2+python+mpi+hdf5", when="@:2.1.1 +catalyst")
-    depends_on("paraview@5.6:+python3+mpi+hdf5", when="@3: +catalyst")
-    depends_on("visit", when="+libsim")
-    depends_on("vtk", when="+libsim")
+    depends_on("paraview@5.6:5.7+python3+mpi+hdf5", when="@3:3.2.1 +catalyst")
+    depends_on("paraview+mpi+python3+hdf5", when="+catalyst")
+    depends_on("visit~gui~python", when="+libsim")
+    depends_on("vtk@8.1.0:8.1.2", when="+libsim")
     depends_on("vtk", when="~libsim ~catalyst")
+    depends_on("vtk+python", when="~libsim ~catalyst+python")
     depends_on("adios", when="+adios")
-    depends_on("hdf5", when="+hdf5")
+    # VTK needs +hl and currently spack cannot resolve +hl and ~hl
+    depends_on("hdf5+hl", when="+hdf5")
     # SENSEI 3 supports Python 3, earlier versions upport only Python 2
     depends_on("python@:2.7.16", when="@:2.1.1 +python", type=('build', 'run'))
```

Comment:
You'll also need to add the following to make sure the python runtime is appropriately setup:
`    extends('python', when='+python')`

Reply:
Thanks Chuck. I added that.